src/targets/shell/curl/client.ts

```diff
@@ -0,0 +1,134 @@
+/**
+ * @description
+ * HTTP code snippet generator for the Shell using cURL.
+ *
+ * @author
+ * @AhmadNassri
+ *
+ * for any questions or issues regarding the generated code snippet, please open an issue mentioning the author.
+ */
+
+import { CodeBuilder } from '../../../helpers/code-builder';
+import { getHeaderName } from '../../../helpers/headers';
+import { quote } from '../../../helpers/shell';
+import { Client } from '../../targets';
+
+export interface CurlOptions {
+  short?: boolean;
+  binary?: boolean;
+  globOff?: boolean;
+}
+
+export const curl: Client<CurlOptions> = {
+  info: {
+    key: 'curl',
+    title: 'cURL',
+    link: 'http://curl.haxx.se/',
+    description: 'cURL is a command line tool and library for transferring data with URL syntax',
+  },
+  convert: ({ fullUrl, method, httpVersion, headersObj, allHeaders, postData }, options) => {
+    const opts = {
+      indent: '  ',
+      short: false,
+      binary: false,
+      globOff: false,
+      ...options,
+    };
+    const { push, join } = new CodeBuilder({
+      indent: opts.indent,
+      // @ts-expect-error SEEMS LEGIT
```

Comment:
What's the error here?

Reply:
![Screenshot_20220413_161929](https://user-images.githubusercontent.com/15232461/163263511-1b16ad9a-ddb3-4749-a9d5-c1e99eacd3a1.png)

This is the problem.  For better or for worse (let's be serious... FOR WORSE, haha) the curl command allowed an option of type boolean to be passed before.

Reply:
see https://github.com/Kong/httpsnippet/pull/248#discussion_r849869109